examples/counter_router/src/main.rs

```diff
@@ -0,0 +1,106 @@
+#[macro_use]
+extern crate stdweb;
+#[macro_use]
+extern crate yew;
+
+use yew::prelude::*;
+use yew::services::console::ConsoleService;
+use yew::services::router::{RouteInfo, RouterTask};
+
+use stdweb::web::Date;
+
+struct Context {
+    console: ConsoleService,
+}
+
+struct Model {
+    value: i64,
+    router: RouterTask<Context, Model>,
+}
+
+enum Msg {
+    Increment,
+    Decrement,
+    None,
+    Bulk(Vec<Msg>),
+}
+
+impl Component<Context> for Model {
+    type Msg = Msg;
+    type Properties = ();
+
+    fn create(_: Self::Properties, context: &mut Env<Context, Self>) -> Self {
+        Model {
+            value: 0,
+            router: RouterTask::new(context, &handle_route),
+        }
+    }
+
+    fn update(&mut self, msg: Self::Msg, context: &mut Env<Context, Self>) -> ShouldRender {
+        match msg {
+            Msg::Increment => {
+                self.value = self.value + 1;
+                context.console.log("plus one");
+            }
+            Msg::Decrement => {
+                self.value = self.value - 1;
+                context.console.log("minus one");
+            }
+            Msg::Bulk(list) => for msg in list {
+                self.update(msg, context);
+                context.console.log("Bulk action");
+            },
+            Msg::None => {
+                context.console.log("No action");
+                return false;
+            }
+        }
+        true
+    }
+}
+
+impl Renderable<Context, Model> for Model {
+    fn view(&self) -> Html<Context, Self> {
+        html! {
+            <div>
+                <p>
+                    <span>{ "Use the buttons below or go to "}</span>
+                    <a href="#increment",>{"#increment"}</a>
+                    <span>{" or "}</span>
+                    <a href="#decrement",>{"#decrement"}</a>
```

Comment:
Ya, I just used the counter since it was a _simple_ example, not necessarily a _good_ one. I wouldn't merge it as an example as-is.

As for clicking multiple times not doing anything... that is how `href` attributes work -- so it isn't a bug in the router, just in the example. What I should do is make the "links" actually buttons and use `router.set_state`. I haven't gotten to that yet though.